frontend/packages/ceph-storage-plugin/src/components/create-storage-system/external-storage/red-hat-ceph-storage/index.tsx

```diff
@@ -0,0 +1,153 @@
+import { CanGoToNextStep, RHCSState, CreatePayload, ExternalComponentProps } from '../types';
+
+import './index.scss';
+
+const SCRIPT_NAME = 'ceph-external-cluster-details-exporter.py';
+
+export const ConnectionDetails: React.FC<ExternalComponentProps<RHCSState>> = ({
+  setFormState,
+  formState,
+}) => {
+  const [pods, podsLoaded, podsLoadError] = useK8sGet<ListKind<PodKind>>(PodModel);
+
+  const { t } = useTranslation();
+
+  React.useEffect(() => {
+    if (podsLoaded && !podsLoadError && pods.items.length) {
+      const address = pods.items?.[0]?.status.podIP;
+      const ipFamily = getIPFamily(address);
+      setFormState('ipFamily', ipFamily);
+    }
+  }, [pods, podsLoadError, podsLoaded, setFormState]);
+
+  const downloadFile = 'download-file.py';
+
+  const handleFileChange: FileUploadProps['onChange'] = (value, fileName) => {
+    setFormState('fileName', fileName);
+    setFormState('fileData', value as string);
+  };
+
+  const handleFileRejected: FileUploadProps['dropzoneProps']['onDropRejected'] = (
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _rejectedFiles,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _event,
+  ) => setFormState('isRejected', true);
+
+  const handleFileAccepted: FileUploadProps['dropzoneProps']['onDropAccepted'] = (
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _acceptedFiles,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    _event,
+  ) =>
+    checkError(formState.fileData, null, null, formState.ipFamily)
+      ? setFormState('isRejected', true)
+      : null;
+
+  return (
+    <FormGroup
+      label={t('ceph-storage-plugin~External storage system metadata')}
+      fieldId="external-storage-system-metadata"
+      className="odf-connection-details__form-group"
+      helperText={
+        <>
+          <Trans t={t} ns="ceph-storage-plugin">
+            Download <code>{{ SCRIPT_NAME }}</code> script and run on the RHCS cluster, then upload
+            the results (JSON).
+          </Trans>{' '}
+          {downloadFile && (
+            <a
+              id="downloadAnchorElem"
+              href={downloadFile}
+              download="ceph-external-cluster-details-exporter.py"
+              target="_blank"
+              rel="noopener noreferrer"
+              className=""
+            >
+              {t('ceph-storage-plugin~Download script')}
+            </a>
+          )}
+        </>
+      }
+      helperTextInvalid={t('ceph-storage-plugin~The uploaded file is not a valid JSON file')}
+      validated={formState.isRejected ? 'error' : 'default'}
+    >
+      <FileUpload
+        id="external-storage-system-metadata"
+        type="text"
+        isRequired
+        className="odf-connection-details__file-upload"
+        browseButtonText={t('ceph-storage-plugin~Browse')}
+        filenamePlaceholder={t('ceph-storage-plugin~Upload script')}
```

Comment:
```suggestion
        filenamePlaceholder={t('ceph-storage-plugin~Upload JSON file')}
```
Are we uploading a script or the file after running the script ?

Reply:
I agree here. Its JSON file.